// @boostxyz/sdk@0.0.0-alpha.20/dist/Boost.js
import { zeroAddress as H, encodeAbiParameters as U, parseAbiParameters as N } from "viem";
var O = typeof globalThis < "u" ? globalThis : typeof window < "u" ? window : typeof global < "u" ? global : typeof self < "u" ? self : {}, E = { exports: {} };
(function(h, l) {
  (function(g, w) {
    h.exports = w(g, 1), l.LibZip = h.exports.LibZip, l.ERC1967Factory = h.exports.ERC1967Factory;
  })(typeof window < "u" ? window : O, function(g, w) {
    var y = {}, c = {};
    y.LibZip = c;
    function x(t) {
      if ((typeof t == "string" || t instanceof String) && (t = t.match(/^[\s\uFEFF\xA0]*(0[Xx])?([0-9A-Fa-f]*)[\s\uFEFF\xA0]*$/))) {
        if (t[2].length % 2)
          throw new Error("Hex string length must be a multiple of 2.");
        return t[2];
      }
      throw new Error("Data must be a hex string.");
    }
    function b(t) {
      return (t | 256).toString(16).slice(1);
    }
    function m(t, r) {
      return parseInt(t.substr(r, 2), 16);
    }
    function L(t) {
      for (var r = [], e = 0; e < t.length; e += 2) r.push(m(t, e));
      return r;
    }
    function P(t) {
      for (var r = "0x", e = 0; e < t.length; r += b(t[e++])) ;
      return r;
    }
    c.flzCompress = function(t) {
      var r = L(x(t)), e = r.length - 4, n = [], i = [], a = 0, o = 2, s = 0, f, v, S, p, z, u, F, Z, T, D;
      function A(d) {
        return r[d] | r[++d] << 8 | r[++d] << 16;
      }
      function B(d) {
        return 2654435769 * d >> 19 & 8191;
      }
      function R(d, j) {
        for (; d >= 32; ) for (i[s++] = 31, f = 32; f--; d--) i[s++] = r[j++];
        if (d) for (i[s++] = d - 1; d--; ) i[s++] = r[j++];
      }
      for (; o < e - 9; ) {
        do
          F = n[S = B(v = A(o))] || 0, z = (p = (n[S] = o) - F) < 8192 ? A(F) : 16777216;
        while (o < e - 9 && o++ && v != z);
        if (o >= e - 9) break;
        for (--o > a && R(o - a, a), u = 0, Z = F + 3, T = o + 3, D = e - T; u < D; u++) D *= r[Z + u] === r[T + u];
        for (o += u, --p; u > 262; u -= 262) i[s++] = 224 + (p >> 8), i[s++] = 253, i[s++] = p & 255;
        u < 7 ? (i[s++] = (u << 5) + (p >> 8), i[s++] = p & 255) : (i[s++] = 224 + (p >> 8), i[s++] = u - 7, i[s++] = p & 255), n[B(A(o))] = o++, n[B(A(o))] = o++, a = o;
      }
      return R(e + 4 - a, a), P(i);
    }, c.flzDecompress = function(t) {
      for (var r = L(x(t)), e = 0, n = 0, i, a, o, s, f = []; e < r.length; )
        if (o = r[e] >> 5)
          for (a = 256 * (r[e] & 31) + r[e + 2 - (o = o < 7)], i = o ? 2 + (r[e] >> 5) : 9 + r[e + 1], e = e + 3 - o, s = n - a - 1; i--; ) f[n++] = f[s++];
        else
          for (i = 1 + r[e++]; i--; ) f[n++] = r[e++];
      return P(f);
    }, c.cdCompress = function(t) {
      t = x(t);
      var r = "0x", e = 0, n = 0, i = 0, a;
      function o(f) {
        r += b((r.length < 4 * 2 + 2) * 255 ^ f);
      }
      function s(f, v) {
        o(0), o(v - 1 + f * 128);
      }
      for (; i < t.length; i += 2) {
        if (a = m(t, i), !a) {
          n && (s(1, n), n = 0), ++e === 128 && (s(0, 128), e = 0);
          continue;
        }
        if (a === 255) {
          e && (s(0, e), e = 0), ++n === 32 && (s(1, 32), n = 0);
          continue;
        }
        n && (s(1, n), n = 0), e && (s(0, e), e = 0), o(a);
      }
      return n && (s(1, n), n = 0), e && (s(0, e), e = 0), r;
    }, c.cdDecompress = function(t) {
      t = x(t);
      for (var r = "0x", e = 0, n, i, a; e < t.length; ) {
        if (i = (e < 4 * 2) * 255 ^ m(t, e), e += 2, !i) {
          for (i = (e < 4 * 2) * 255 ^ m(t, e), a = (i & 127) + 1, e += 2, n = 0; n < a; ++n) r += b((i >> 7 && n < 32) * 255);
          continue;
        }
        r += b(i);
      }
      return r;
    };
    var C = {};
    return y.ERC1967Factory = C, C.address = "0x0000000000006396FF2a80c067f99B3d2Ab4Df24", C.abi = JSON.parse('[{0:[],1:"DeploymentFailed"96"SaltDoesNotStartWithCaller"96"Unauthorized"96"UpgradeFailed",2:3959790,9791],1:"AdminChanged",2:10959790,9792,9791],1:"Deployed",2:10959790,9792],1:"Upgraded",2:10},{0:[{90],1:"adminOf",12:[{9199{0:[{90,{91],1:"changeAdmin",12:[],13:"nonpayable",2:15},{0:[{92,{91],1:"deploy",12:[{9098,{0:[{92,{91,{94],1:"deployAndCall",12:[{9098,{0:[{92,{91,{93],1:"deployDeterministic",12:[{9098,{0:[{92,{91,{93,{94],1:"deployDeterministicAndCall",12:[{9098,{0:[],1:"initCodeHash",12:[{6:19,1:"result",2:19}99{0:[{93],1:"predictDeterministicAddress",12:[{6:7,1:"predicted",2:7}99{0:[{90,{92],1:"upgrade",12:[98,{0:[{90,{92,{94],1:"upgradeAndCall",12:[98]'.replace(/9\d/g, function(t) {
      return ["6:7,1:8,2:7}", "6:7,1:9,2:7}", "6:7,1:11,2:7}", "6:19,1:20,2:19}", "6:17,1:18,2:17}", "},{4:false,0:[", ",2:3},{0:[],1:", "{5:true,", "],13:16,2:15}", "],13:14,2:15},"][t - 90];
    }).replace(/\d+/g, function(t) {
      return '"' + "inputs,name,type,error,anonymous,indexed,internalType,address,proxy,admin,event,implementation,outputs,stateMutability,view,function,payable,bytes,data,bytes32,salt".split(",")[t] + '"';
    })), w || (g.solady = y), y;
  });
})(E, E.exports);
var k = E.exports;
class G {
  /**
   * Creates an instance of Boost.
   *
   * @constructor
   * @param {BoostConfig} config
   */
  constructor(l) {
    this.id = l.id, this.budget = l.budget, this.action = l.action, this.validator = l.validator, this.allowList = l.allowList, this.incentives = l.incentives, this.protocolFee = l.protocolFee || 0n, this.referralFee = l.referralFee || 0n, this.maxParticipants = l.maxParticipants || 0n, this.owner = l.owner || H;
  }
}
function I({
  budget: h,
  action: l,
  validator: g,
  allowList: w,
  incentives: y,
  protocolFee: c = 0n,
  referralFee: x = 0n,
  maxParticipants: b = 0n,
  owner: m
}) {
  return k.LibZip.cdCompress(
    U(
      N([
        "BoostPayload payload",
        "struct BoostPayload { address budget; Target action; Target validator; Target allowList; Target[] incentives; uint64 protocolFee; uint64 referralFee; uint256 maxParticipants; address owner; }",
        "struct Target { bool isBase; address instance; bytes parameters; }"
      ]),
      [
        {
          budget: h,
          action: l,
          validator: g,
          allowList: w,
          incentives: y,
          protocolFee: c,
          referralFee: x,
          maxParticipants: b,
          owner: m
        }
      ]
    )
  );
}
export {
  G as Boost,
  I as prepareBoostPayload
};
